packages/core/core/src/middlewares/security.ts

```diff
@@ -71,12 +71,24 @@ export const security: Common.MiddlewareFactory<Config> =
       });
     }
 
-    if (ctx.method === 'GET' && ['/admin'].some((str) => ctx.path.startsWith(str))) {
+    /**
+     * These are for vite's watch mode so it can accurately
+     * connect to the HMR websocket & reconnect on failure
+     * or when the server restarts.
+     *
+     * It only applies in development, and only on GET requests
+     * that are part of the admin route.
+     */
+    if (
+      process.env.NODE_ENV === 'development' &&
```

Comment:
Improvement idea: 

Instead of touching any of the internal why not use sth similar to vite-express that adds the necessary endpoints for vite to the router outside of the middleware stack so we don't have to take any chances when serving vite files.

Reply:
That would be really cool, i'd be interested working with @innerdvations on that for sure.